skip reassign SSH port only when on hyperv/system networking

lstocchi opened the pull request:
PR https://github.com/cfergeau/podman/pull/19/files introduced a check to only verify/reassign busy ssh port when using user-mode networking.

This was needed for when using hyperv with system networking mode because we set a custom IP/use default ssh port and do not need to verify that port 22 is busy on the host.

However this creates a misbehavior on WSL when using mirrored mode https://github.com/crc-org/macadam/issues/289

This patch just reinforces the local port check to specify the hypervisor we target (hyperv).

All other hypervisors rely on usermode networking, and WSL is a special case.

cfergeau commented on pkg/machine/shim/networking.go:
```diff
@@ -98,7 +98,7 @@ func startHostForwarder(mc *vmconfigs.MachineConfig, provider vmconfigs.VMProvid
 func startNetworking(mc *vmconfigs.MachineConfig, provider vmconfigs.VMProvider) (string, machine.APIForwardingState, error) {
 	// Check if SSH port is in use, and reassign if necessary
-	if provider.UserModeNetworkEnabled(mc) && !ports.IsLocalPortAvailable(mc.SSH.Port) {
+	if (provider.VMType() != define.HyperVVirt || provider.UserModeNetworkEnabled(mc)) && !ports.IsLocalPortAvailable(mc.SSH.Port) {
```
On WSL + mirrored mode, `provider.UserModeNetworkEnabled(mc)` will be false?

lstocchi replied:
yes, there should be no way to enable it if we want to rely on mirrored mode so it should always return false

cfergeau replied:
Wondering if the check should be `provider.VMType() == define.WSLVirt`? With `!= define.HyperVVirt`, it will trigger for (hypothetical) system networking scenarios with vfkit or qemu.

cfergeau replied:
Since more changes will be needed for mirrored mode, I’ll leave it out of the upcoming macadam release, we can include it slightly later.

lstocchi replied:
Updated using your suggestion on WSLVirt check

cfergeau commented:
Can you rebase this on top of the 5.7 branch? I think we can move forward with it now.

lstocchi committed:
Reassign SSH port also when on WSL do not use usermode networking

PR
https://github.com/cfergeau/podman/pull/19/files
introduced a check to only verify/reassign busy ssh port when using user-mode
networking.

This was needed for when using hyperv with system networking
mode because we set a custom IP/use default ssh port and do not need to verify that port 22
is busy on the host.

However this creates a misbehavior on WSL when using mirrored mode
crc-org/macadam#289

This patch just reinforces the local port check to specify that WSL
needs to be verified even when using system networking.

Signed-off-by: lstocchi <lstocchi@redhat.com>

lstocchi commented:
> Can you rebase this on top of the 5.7 branch? I think we can move forward with it now.

Done
